options/logger.go

```diff
@@ -82,190 +56,171 @@ func (f LogFormat) MakeFormatter() (send.MessageFormatter, error) {
 	}
 }
 
-// Buffer packages options for whether or not a Logger should be
-// buffered and the duration and size of the respective buffer in the case that
-// it should be.
-type Buffer struct {
-	Buffered bool          `bson:"buffered" json:"buffered" yaml:"buffered"`
-	Duration time.Duration `bson:"duration" json:"duration" yaml:"duration"`
-	MaxSize  int           `bson:"max_size" json:"max_size" yaml:"max_size"`
+// LogConfig type aliases []byte in order to implement the bson.MarshalBSON and
+// json.MarshalJSON interfaces. It is intended to hold the logger configuration
+// data.
+type LogConfig []byte
+
+func (lc LogConfig) MarshalBSON() ([]byte, error) { return lc, nil }
+func (lc LogConfig) MarshalJSON() ([]byte, error) { return lc, nil }
+
+// LoggerConfig represents the necessary information to construct a new grip
+// send.Sender. LoggerConfig implements the bson.MarshalBSON and
+// json.MarshalJSON interfaces.
+type LoggerConfig struct {
+	Type     string          `json:"type" bson:"type"`
+	Format   LogConfigFormat `json:"format" bson:"format"`
+	Config   LogConfig       `json:"config" bson:"config"`
+	Registry LoggerRegistry  `json:"-" bson:"-"`
+
+	producer LoggerProducer
+	sender   send.Sender
 }
 
-// Validate ensures that BufferOptions is valid.
-func (opts Buffer) Validate() error {
-	if opts.Buffered && opts.Duration < 0 || opts.MaxSize < 0 {
-		return errors.New("cannot have negative buffer duration or size")
+// Validate ensures the LoggerConfig is valid.
+func (lc *LoggerConfig) Validate() error {
+	catcher := grip.NewBasicCatcher()
+
+	catcher.NewWhen(lc.Type == "", "cannot have empty logger type")
+	catcher.Add(lc.Format.Validate())
+
+	if lc.Registry == nil {
+		lc.Registry = globalLoggerRegistry
```

Comment:
It seems kind of weird that every single LoggerConfig instance contains the global logging registry. Is there reason we can't just get rid of this field and just get logger factories directly from the global registry?